samples/sample-07/src/test/java/com/example/sample07/Sample07ApplicationTest.java

```diff
@@ -0,0 +1,335 @@
+		config.partitionAssignedLatch = new CountDownLatch(1);
+		config.listenerLatch = new CountDownLatch(1);
+		rawConsumerPollCount = new CountDownLatch(1);
+
+		// If kafka consumer is created, consumer backed ground thread will send group coordinator heartbeat request to broker.
+		final KafkaConsumer<Object, String> rawConsumer = createRawConsumer();
+
+		// When consumer subscribe topic, broker aware it.
+		// And consumer will get assignment as response of poll.
+		rawConsumer.subscribe(List.of(TOPIC_NAME), new RawConsumerRebalanceListener());
+		threadPoolTaskExecutor.execute(() -> {
+			while (true) {
+				ConsumerRecords<Object, String> poll = rawConsumer.poll(Duration.ofSeconds(1));
+				if (!poll.isEmpty()) {
+					rawConsumerPollCount.countDown();
+					break;
+				}
+			}
+		});
+
+		// spring-kafka consumer will revoke one of partitions.
+		// because two topic partitions, two consumers.
+		assertThat(rawConsumerAssignedCount.await(20, TimeUnit.SECONDS)).isTrue();
+		assertThat(config.partitionRevokedLatch.await(20, TimeUnit.SECONDS)).isTrue();
+		assertThat(config.partitionAssignedLatch.await(20, TimeUnit.SECONDS)).isTrue();
+
+		// Send two messages each partition to broker.
+		this.template.send(TOPIC_NAME, 0, null, "my-data");
+		this.template.send(TOPIC_NAME, 1, null, "my-data");
+
+		// Both spring-kafka consumer and rawConsumer will received message from broker each by each.
+		assertThat(config.listenerLatch.await(20, TimeUnit.SECONDS)).isTrue();
+		assertThat(rawConsumerPollCount.await(20, TimeUnit.SECONDS)).isTrue();
+
+		// Set initial CountDown latch.
+		// Scenario : one consumer closed, the other will have two partitions as result of rebalancing.
+		rawConsumerRevokedCount = new CountDownLatch(1);
+		config.partitionAssignedLatch = new CountDownLatch(1);
+
+		rawConsumer.close();
+
+		assertThat(rawConsumerRevokedCount.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(config.partitionAssignedLatch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		config.listenerLatch = new CountDownLatch(2);
+		this.template.send(TOPIC_NAME, 0, null, "my-data");
+		this.template.send(TOPIC_NAME, 1, null, "my-data");
+		assertThat(config.listenerLatch.await(20, TimeUnit.SECONDS)).isTrue();
+	}
+
+	KafkaConsumer<Object, String> createRawConsumer() {
+		final Map<String, Object> props = getConsumerProperties();
+		return new KafkaConsumer<>(props);
+	}
+
+	static Map<String, Object> getConsumerProperties() {
+		final Map<String, Object> props = new HashMap<>();
+
+		props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+		props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+		props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BROKER);
+		props.put(ConsumerConfig.GROUP_ID_CONFIG, GROUP_ID);
+		props.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, "consumer");
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
```

Comment:
These all could go to the `application.properties`

Reply:
This is not for consumer managed by `spring-kafka`. 

I make two `consumers` which not are managed by `spring-kafka`.
One is `rawConsumer` for triggering `consumer rebalance`. 
The other one is `dummyConsumer` for making broker scheduling to load metadata. (https://github.com/spring-projects/spring-kafka/pull/3237#discussion_r1596833689)

Thus, we don't need to `application.properties`. 
I think `application.properties` is for the consumer managed by `spring-kafka`, right?  (for example, `spring.kafka.admin.client-id`) 
If you want to migrate this to properties, how about making `dummy-consumer.properties` and use it? 
```java
	static Map<String, String> getPropertiesFromFile() {
		final Resource resource = new ClassPathResource(BROKER_PROPERTIES_FILE_PATH);
		try {
			final Properties properties = PropertiesLoaderUtils.loadAllProperties(resource.getFilename());
			return properties.stringPropertyNames().stream()
							.collect(Collectors.toMap(
										s -> s,
										s -> (String) properties.get(s)));
		}
		catch (IOException e) {
			throw new RuntimeException(e);
		}
	}
```

Reply:
Ah, i missed one. 
I will move the settings for consumer which are managed by `spring-kafka` to `application.properties`. However, it seems necessary to leave the settings for `dummyConsumer` and `rawConsumer` as they are.

What do you think?